### DISCUSSION TITLE
fmltc Setup instructions - "Install JDK" step

### DISCUSSION THREAD

**Discussion by BalaKasi** [FIRST-Tech-Challenge/fmltc#224] [Q&A] [answered]
In the fmltc Setup instructions, after the "Install JDK" step one is expected to "Fill in the values in server/env_variables.yaml".  Is the "Install JDK" supposed to create a file named env_variables.yaml in the /server folder?  Since this file is nowhere to be found, it would help if an example is provided to show the environment variable that need to be set.

**Comment by lizlooney** [comment, marked as answer]
Did you clone the FIRST-Tech-Challenge/fmltc?
That should actually be the first step, so I'll add that.

**Comment by lizlooney** [comment]
I've updated the README.md file. https://github.com/FIRST-Tech-Challenge/fmltc/commit/d75dcf22ffc4487d1a83eb9379615c4f144b9b73

**Reply by BalaKasi** [reply]
Very good and thanks!  I located and completed the "PROJECT-ID" edit to env_variables.yaml.  On the base URL edit, I ran into this problem - I don't have a base URL yet since I haven't deployed the appengine yet.  Does this mean I follow the rest of steps and then come back to edit this env variable?

**Reply by cmacfarl** [reply]
That'll work.  Just remember to redeploy the app engine server after you edit env_variables.yaml.

**Reply by BalaKasi** [reply]
When one reaches the "Setup the environment" step, are there prereqs like ensuring the bash commands are available in a Windows 10 or 11 machine?  The source command is not a Windows command.

**Reply by cmacfarl** [reply]
Yeah, I think it's beyond the scope of the project to provide explicit instructions for all three major platforms (two if you consider Mac's BSD derivative to be equivalent to Linux).  Maybe a disclaimer could be put at the top that the following instructions eventually require a bash shell.  At any rate, I highly recommend [GitForWindows](https://gitforwindows.org/).  It comes packaged with a very nice bash shell that will run all the commands in the setup instructions.

**Reply by BalaKasi** [reply]
I understand and thanks.  GitForWindows certainly helped; only snag is that it doesn't support the colrm filter and so the "column remove" has to be done with alternate syntax.

**Comment by lizlooney** [comment]
I think if you go to https://console.cloud.google.com/iam-admin/iam?project=my_project_id (replace my_project_id with your actual project ID) and look at the upper right, it will show you your URL, even if you haven't deployed to appengine yet.

**Reply by BalaKasi** [reply]
I checked before deploying to appengine but didn't see the URL; I'll check again AFTER I deploy to appengine.

**Reply by cmacfarl** [reply]
It used to be that the url was just \<project-name>.appspot.com, then Google decided to insert region tokens into the domain.  So the url now depends upon where you deploy the server.  If you know Google's pattern for region tokens you can guess what the url will be, but it's likely just easier to deploy, go grab the url, edit the yaml and redeploy.

**Reply by BalaKasi** [reply]
Correct - I used northamerica-northeast1 and got <project-name>.nn.r.appspot.com.  It would be difficult to glean Google's pattern for region tokens.
